### PR TITLE
user_activity

### PR DESCRIPTION
**Pull request by phamkykhoi**

### CODE REVIEW

File: app/controllers/followers_controller.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class FollowersController < ApplicationController
+	before_action :authenticate_user!
```
Comment (PhamVanTung) [inline app/controllers/followers_controller.rb]: indent

File: app/controllers/followings_controller.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 class FollowingsController < ApplicationController
+	before_action :authenticate_user!
```
Comment (PhamVanTung) [inline app/controllers/followings_controller.rb]: indent

File: app/views/shared/_show_user.html.erb [hunk 3]
```diff
@@ -18,5 +18,12 @@
         <span class="label label-default">Lesson</span>
       <% end %>
     </h1>
+    <br/>
+    <h3>Activities<hr/></h3>
+      <ul class="lesson-categories">
+        <% if @activities != nil %>
```
Comment (dieunb) [inline app/views/shared/_show_user.html.erb]: `unless @activities.blank?`

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -11,7 +11,7 @@ def create
       params[:session][:remember_me] == '1' ? remember(@user)  : forget(@user)
```
Comment (PhamVanTung) [inline app/controllers/sessions_controller.rb]: remember(@user)  : forget(@user) bỏ 1 dấu space

File: app/controllers/static_pages_controller.rb [hunk 5]
```diff
@@ -2,5 +2,8 @@ class StaticPagesController < ApplicationController
   before_action :authenticate_user!
   
   def home
+    @user = User.find current_user
+    @user_lessons = Lesson.lesson_user current_user
+    @activities = current_user.activities
   end
-end
+end
```
Comment (PhamVanTung) [inline app/controllers/static_pages_controller.rb]: add new line

File: app/models/activity.rb [hunk 6]
```diff
@@ -7,6 +7,6 @@ class Activity < ActiveRecord::Base
   validates :target_id, presence: true
 
   def return_target_object
-    lesson? ? Lesson.find(target_id) : User.find(target_id)
+    learned? ? Lesson.find(target_id) : User.find(target_id)
   end
-end
+end
```
Comment (PhamVanTung) [inline app/models/activity.rb]: add new line

File: app/models/relationship.rb [hunk 7]
```diff
@@ -13,10 +13,10 @@ class Relationship < ActiveRecord::Base
...
   def follow_activity
-    create_activity follower_id, followed_id , "follow"
+    create_activity follower_id, followed_id , "followed"
   end
 
   def unfollow_activity
-    create_activity follower_id, followed_id , "unfollow"
+    create_activity follower_id, followed_id , "unfollowed"
   end
 end
```
Comment (PhamVanTung) [inline app/models/relationship.rb]: add new line

File: app/views/activities/_activity.html.erb [hunk 8]
```diff
@@ -1,5 +1,6 @@
 <% target = activity.return_target_object %>
-<% if activity.lesson? %>
+
+<% if activity.learned? %>
    <li>
```
Comment (PhamVanTung) [inline app/views/activities/_activity.html.erb]: indent

File: app/controllers/static_pages_controller.rb [hunk 9]
```diff
@@ -2,5 +2,9 @@ class StaticPagesController < ApplicationController
   before_action :authenticate_user!
   
   def home
+    @user = User.find current_user
```
Comment (dieunb) [inline app/controllers/static_pages_controller.rb]: i don't understand you

File: app/controllers/users_controller.rb [hunk 10]
```diff
@@ -22,8 +22,19 @@ def create
   end
 
   def show
-    @user = User.find params[:id]
-    @user_lessons = Lesson.lesson_user @user
+    @user  = @user = User.friendly.find params[:id]
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: ???

File: config/routes.rb [hunk 11]
```diff
@@ -67,10 +67,7 @@
   root "static_pages#home"
   devise_for :users
   
-  resources :users, except: [:edit, :update] do
-    resources :followers, only: :index
-    resources :followings, only: :index
-  end
+  resources :users, except: [:edit, :update]
```
Comment (dieunb) [inline config/routes.rb]: i want to see followings and followers with url `/users/1/followings` or `/users/1/followers`

File: config/routes.rb [hunk 12]
```diff
@@ -68,8 +68,7 @@
   devise_for :users
   
   resources :users, except: [:edit, :update] do
-    resources :followers, only: :index
-    resources :followings, only: :index
+    get ":follow" => 'users#show', as: :show_follow
```
Comment (dieunb) [inline config/routes.rb]: I want to handle it at relations_controller

File: app/controllers/users_controller.rb [hunk 13]
```diff
@@ -22,8 +22,21 @@ def create
   end
 
   def show
-    @user = User.find params[:id]
-    @user_lessons = Lesson.lesson_user @user
+    if params[:follow] == "following"
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: refactor to clear code

File: app/controllers/relationships_controller.rb [hunk 14]
```diff
@@ -1,4 +1,21 @@
 class RelationshipsController < ApplicationController
+
+  def index
+    if %w(followings followers).include? params[:follow]
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: move setting

File: app/controllers/relationships_controller.rb [hunk 15]
```diff
@@ -1,4 +1,21 @@
...
+      @title = params[:follow] == "followings" ? 
+        (I18n.t "view.following") : (I18n.t "view.follower")
+      
+      @user  = @user = User.find params[:user_id]
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: `@user  = @user`

File: app/controllers/static_pages_controller.rb [hunk 16]
```diff
@@ -2,5 +2,9 @@ class StaticPagesController < ApplicationController
   before_action :authenticate_user!
   
   def home
+  	@user = current_user
```
Comment (dieunb) [inline app/controllers/static_pages_controller.rb]: remove

**Pull request merged**